sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -877,9 +877,14 @@ def collect_index_usage_metrics(self):
         if not self._index_usage_last_check_ts or now - self._index_usage_last_check_ts > interval:
             self._index_usage_last_check_ts = now
             self.log.debug('Collecting index usage statistics')
-            db_names = [d.name for d in self.databases] or [
-                self.instance.get('database', self.connection.DEFAULT_DATABASE)
-            ]
+            default_database = self.instance.get('database', self.connection.DEFAULT_DATABASE)
+            # Filter out tempdb as the query might be blocking and it's index usage information is not relevant
+            include_temdb_in_metric = is_affirmative(self.instance.get('include_index_usage_metrics_tempdb', False))
+
+            db_names = [d.name for d in self.databases if include_temdb_in_metric or d != 'tempdb'] or (
+                [default_database] if include_temdb_in_metric or default_database != 'tempdb' else []
+            )
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** i found this a bit hard to understand. maybe do something like
```python
db_names = [d.name for d in self.databases] or [
    self.instance.get('database', self.connection.DEFAULT_DATABASE)
]
if not include_index_usage_metrics_tempdb:
    db_names = [db_name for db_name in db_names if db_name != 'tempdb']
```
‏

Reply:
Agreed, will do.